assignment/impl/src/java/org/sakaiproject/assignment/impl/NodeToStringUtility.java

```diff
@@ -0,0 +1,32 @@
+package org.sakaiproject.assignment.impl;
+
+import java.io.StringWriter;
+
+import javax.xml.XMLConstants;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Node;
+
+public class NodeToStringUtility {
+
+    private NodeToStringUtility() {
+
+    }
+
+    public static String toString(Node node) throws TransformerException {
+        final TransformerFactory tf = TransformerFactory.newInstance();
+        tf.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
```

Comment:
Shouldn't this be set on the entire xml vs just the assignments node? Does it make sense for each tool to have to do this? I would look where the xml document is created and just set it there, there are a number of examples in Sakai that already do this i.e.
`org.sakaiproject.citation.impl.BaseConfigurationService#getXmlDocumentBuilder`

Reply:
I think this would be beyond this PR. We can create a new JIRA and assign ourselves and do that refactor once this is working, can't we?

Reply:
Agree it should be a new JIRA